### PR TITLE
feat: support aggr filter for flexgroup volumes

### PR DESCRIPTION
**Pull request by Hardikl**
UI panels:
image

This panel has 2 info to be shown, second info is from ontap.
image

new metric from zapiperf:
image

metric labels for flexgroup:
image

### CODE REVIEW

File: grafana/dashboards/cmode/aggregate.json [hunk 1]
```diff
@@ -3792,6 +3792,389 @@
...
+        "h": 1,
+        "w": 24,
+        "x": 0,
+        "y": 81
+      },
+      "id": 81,
+      "panels": [
+        {
+          "datasource": "${DS_PROMETHEUS}",
+          "description": "For FlexGroup, This panel would show a single consolidated value even when one aggregate is selected in the dropdown.",
```
Comment (cgrinds) [inline grafana/dashboards/cmode/aggregate.json:3809]: maybe something like this is clearer?

> Flexgroup by-aggregate filtering does not display the per-aggregate breakdown, instead the sum of all flexgroup aggregates is displayed. This is how ONTAP reports the data, even when an aggregate is selected in the dropdown.

File: grafana/dashboards/cmode/aggregate.json [hunk 2]
```diff
@@ -3792,6 +3792,389 @@
...
+            {
+              "exemplar": false,
+              "expr": "topk($TopResources, volume_size_used{datacenter=~\"$Datacenter\",cluster=~\"$Cluster\", volume=~\"$TopVolumeSizeUsed\"} * on (cluster, svm, volume) group_left(aggr) volume_aggrs{datacenter=~\"$Datacenter\",cluster=~\"$Cluster\",aggr=~\".*$Aggregate.*\",volume=~\"$TopVolumeSizeUsed\"})",
+              "hide": false,
+              "interval": "",
+              "legendFormat": "{{aggr}} - {{volume}} - {{style}}",
+              "refId": "D"
+            }
+          ],
+          "title": "Volume Space Used",
```
Comment (cgrinds) [inline grafana/dashboards/cmode/aggregate.json:3892]: Volume Space Used by Aggregate

File: grafana/dashboards/cmode/aggregate.json [hunk 3]
```diff
@@ -3792,6 +3792,389 @@
...
+              "refId": "A"
+            }
+          ],
+          "title": "Volume Snapshot Space Used",
+          "transformations": [],
+          "type": "timeseries"
+        },
+        {
+          "datasource": "${DS_PROMETHEUS}",
+          "description": "For FlexGroup, This panel would show a single consolidated value even when one aggregate is selected in the dropdown.\n\nNote that in some scenarios, it is possible to pass 100% of the space allocated.\n",
```
Comment (cgrinds) [inline grafana/dashboards/cmode/aggregate.json:4084]: ...Note that in some scenarios, it is possible to exceed 100% of the space allocated

File: grafana/dashboards/cmode/aggregate.json (hunk 1 above)
Comment (Hardikl) [reply grafana/dashboards/cmode/aggregate.json]: done

**Review by cgrinds — APPROVED** [review]

File: cmd/collectors/zapiperf/plugins/volume/volume.go [hunk 4]
```diff
@@ -65,11 +92,21 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
 			}
 
+			// set aggrs label for flexgroup
+			aggrs := make([]string, 0)
+			for aggr := range flexgroupAggrsMap[key] {
+				aggrs = append(aggrs, aggr)
+			}
+			// make sure the order of aggregate is same for each poll
+			sort.Strings(aggrs)
+			fg.SetLabel("aggr", strings.Join(aggrs, ","))
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/zapiperf/plugins/volume/volume.go:108]: done

File: cmd/collectors/zapiperf/plugins/volume/volume.go [hunk 5]
```diff
@@ -31,28 +32,54 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 	opsKeyPrefix := "temp_"
 	re := regexp.MustCompile(`^(.*)__(\d{4})$`)
 
+	metricName := "aggrs"
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/zapiperf/plugins/volume/volume.go:35]: done

File: cmd/collectors/zapiperf/plugins/volume/volume.go [hunk 6]
```diff
@@ -31,28 +32,54 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
+		me.Logger.Error().Stack().Err(err).Msg("add metric")
+		return nil, err
+	}
+
 	cache := data.Clone(false, true, false)
 	cache.UUID += ".Volume"
 
+	me.Logger.Trace().Msgf("added metric: (%s) [%s] %v", metricName, metricName, metric)
+
+	flexgroupAggrsMap := make(map[string]map[string]bool)
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/zapiperf/plugins/volume/volume.go:47]: done, made the change in other PR: https://github.com/NetApp/harvest/pull/1695

**Pull request merged**